impl-tinkerpop-parent/tinkergraph-provider/src/main/java/org/hawkular/inventory/impl/tinkerpop/provider/TinkerGraphProvider.java

```diff
@@ -39,6 +45,47 @@ public void ensureIndices(WrappedTinkerGraph graph, IndexSpec... indexSpecs) {
         //don't bother with this for a demo graph
     }
 
+    @Override
+    public InventoryBackend.Transaction startTransaction(WrappedTinkerGraph graph, boolean mutating) {
+        ReentrantReadWriteLock lock;
+        synchronized (transactionLocks) {
+            lock = transactionLocks.get(graph);
+            if (lock == null) {
+                lock = new ReentrantReadWriteLock();
+                transactionLocks.put(graph, lock);
+            }
+        }
+
+        if (mutating) {
+            lock.writeLock().lock();
+        } else {
+            lock.readLock().lock();
+        }
+
+        return new SimulatedSerializedTransaction(mutating, lock);
+    }
+
+    @Override
+    public void commit(WrappedTinkerGraph graph, InventoryBackend.Transaction t) {
+        graph.commit();
+        unlock(t);
```

Comment:
perhaps
````java
try {
  graph.commit();
} finally {
  unlock(t);
}
````
.. would be better

What if I start a transaction and forget to call the commit/rollback (or I forgot to place it to the finally clause and exception is thrown). We'll get the deadlock, right? Doing some timeout logic is overkill for the TinkerGraph, perhaps we'll get it for free with the Titan.